utils/pullRequests.js

```diff
@@ -0,0 +1,71 @@
+const { fetchMultiplePageResults } = require("./fetchMultiplePageResults");
+const { getDateTimeRangeForPRs } = require("./helper");
+const githubService = require("../services/githubService");
+
+const getFilteredPRsOrIssues = async (qualifiers) => {
+  let allPRs = [];
+  const { sortBy = "RECENT_FIRST", filterBy } = qualifiers;
+  const order = sortBy === "RECENT_FIRST" ? ORDER_TYPE.DESC : ORDER_TYPE.ASC;
+
+  const startDate = qualifiers?.startDate;
+  const endDate = qualifiers?.endDate;
+  const dateTime = getDateTimeRangeForPRs(startDate, endDate);
+
+  const searchParams = {}; // searchParams used to create list of params to create github API URL
+  const resultOptions = { order }; // resultOptions is used for ordering, searching within date-time range and pagination of results
+
+  if (filterBy === "OPEN_PRS") {
+    if (dateTime) {
+      searchParams.created = dateTime;
+    }
+
+    allPRs = await fetchMultiplePageResults(githubService.fetchOpenPRs, {
+      searchParams,
+      resultOptions,
+    });
```

Comment:
why are we writing the same thing in all the if blocks can we set the options based on the if block and call the function in the end?

Reply:
changes made, thanks for pointing out